plugins/inputs/postgresql/postgresql.go

```diff
@@ -42,6 +40,20 @@ var sampleConfig = `
   ##
   address = "host=localhost user=postgres sslmode=disable"
 
+  ## connection configuration.
+
+  ## maxidle - specify the number of idle connections to maintain. see sql.DB for details.
+  ## default is 1. setting to 0 to force disconnect after use.
+  max_idle = 1
+
+  ## maxopen - specify the maximum number of connections to maintain. see sql.DB for details.
+  ## default is 1.
+  max_open = 1
```

Comment:
If gather takes too long, it spits out an error and does nothing. It will not launch a second gather. Part of the [plans I had for this](https://github.com/influxdata/telegraf/issues/1977#issuecomment-278446273) was to parallelize the queries. It would also be beneficial to share the pool among multiple instances of the plugin. Then it would serve a purpose.